packages/react/src/transactions/useUpdateFollowPolicy.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  PendingSigningRequestError,
+  ProfileId,
+  TransactionKind,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import {
+  ChargeFollowPolicy,
+  NoFeeFollowPolicy,
+  UpdateFollowPolicyRequest,
+} from '@lens-protocol/domain/use-cases/profile';
+import { useState } from 'react';
+
+import {
+  TransactionState,
+  useWaitUntilTransactionSettled,
+} from './adapters/TransactionQueuePresenter';
+import { useUpdateFollowPolicyController } from './adapters/useUpdateFollowPolicyController';
+
+type UseUpdateFollowPolicyArgs = {
+  profileId: ProfileId;
+  followPolicy: ChargeFollowPolicy | NoFeeFollowPolicy;
+};
+
+export function useUpdateFollowPolicy() {
+  const [error, setError] = useState<
+    PendingSigningRequestError | UserRejectedError | WalletConnectionError | null
+  >(null);
+  const [isPending, setIsPending] = useState<boolean>(false);
+  const updateFollowPolicy = useUpdateFollowPolicyController();
+  const waitUntilTransactionIsSettled = useWaitUntilTransactionSettled();
+
+  return {
+    updateFollowPolicy: async ({ profileId, followPolicy }: UseUpdateFollowPolicyArgs) => {
+      setError(null);
+      setIsPending(true);
+
+      try {
+        const result = await updateFollowPolicy({
+          profileId,
+          policy: followPolicy,
+          kind: TransactionKind.UPDATE_FOLLOW_POLICY,
+        });
+
+        if (result.isFailure()) {
+          setError(result.error);
+        }
+
+        await waitUntilTransactionIsSettled(
+          (transaction): transaction is TransactionState<UpdateFollowPolicyRequest> =>
+            transaction.request.kind === TransactionKind.UPDATE_FOLLOW_POLICY &&
+            transaction.request.profileId === profileId,
+        );
```

Comment:
Do we need to wait? I don't think we necessarily need to have the user waiting, in the end it's about someone else following them. Compared to setting the dispatcher for example I cannot something that the user can do wrong if they we "let them go" when this is not mined/indexed yet.